src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -891,6 +895,22 @@ internal RunVersionOne CreateRun(Run v2Run)
             return run;
         }
 
+        private IDictionary<string, LogicalLocationVersionOne> CreateLogicalLocationsDictionary(IList<LogicalLocation> logicalLocations)
+        {
+            Dictionary<string, LogicalLocationVersionOne> logicalLocationsVersionOne = null;
+
+            if (logicalLocations != null)
+            {
+                logicalLocationsVersionOne = new Dictionary<string, LogicalLocationVersionOne>();
+                foreach (LogicalLocation logicalLocation in logicalLocations)
+                {
+                    logicalLocationsVersionOne[logicalLocation.FullyQualifiedName] = CreateLogicalLocation(logicalLocation);
```

Comment:
>FullyQualifiedName [](start = 63, length = 18)

Again FQLN ?? Name.

You know, this happens often enough that I suggest we add a partial class for `LogicalLocation` that defines a property... I don't know what to call it, "SafeName" or something... that returns FQLN ?? Name, and use that in all the places I mentioned.

Reply:
I think a helper function probably makes most sense. an extension method off LogicalLocation perhaps. LogicalLocation.GetSarifVersionOneLogicalLocationsDictionaryKey. whew

---
In reply to: [243867203](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243867203) [](ancestors = 243867203)